src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_RankSumTest.java

```diff
@@ -99,4 +119,188 @@ public void calculateRawData(VariantContext vc, final ReadLikelihoods<Allele> li
             }
         }
     }
+
+    /**
+     * Unless overridden by client methods, parses the raw data histograms generated for each allele and computes the
+     * median value for each allele and outputs that as the representative double value.
+     *
+     * @param vc -- contains the final set of alleles, possibly subset by GenotypeGVCFs
+     * @param originalVC -- used to get all the alleles for all gVCFs
+     * @return
+     */
+    public  Map<String, Object> finalizeRawData(final VariantContext vc, final VariantContext originalVC) {
+        if (!vc.hasAttribute(getRawKeyName())) {
+            return new HashMap<>();
+        }
+        final String rawRankSumData = vc.getAttributeAsString(getRawKeyName(),null);
+        if (rawRankSumData == null) {
+            return new HashMap<>();
+        }
+        final Map<String,Object> annotations = new HashMap<>();
+        final AlleleSpecificAnnotationData<Histogram> myData = new AlleleSpecificAnnotationData<>(originalVC.getAlleles(), rawRankSumData);
+        parseRawDataString(myData);
+
+        final Map<Allele, Double> perAltRankSumResults = calculateReducedData(myData.getAttributeMap(), myData.getRefAllele());
+        //shortcut for no ref values
+        if (perAltRankSumResults.isEmpty()) {
+            return annotations;
+        }
+        final String annotationString = makeReducedAnnotationString(vc, perAltRankSumResults);
+        annotations.put(getKeyNames().get(0), annotationString);
+        return annotations;
+    }
+
+    // Calculates the median of values per-Allele in the RMS tests
+    public Map<Allele, Double> calculateReducedData(final Map<Allele, Histogram> perAlleleValues, final Allele ref) {
+        final Map<Allele, Double> perAltRankSumResults = new HashMap<>();
+        for (final Allele alt : perAlleleValues.keySet()) {
+            if (!alt.equals(ref, false) && perAlleleValues.get(alt) != null) {
+                perAltRankSumResults.put(alt, perAlleleValues.get(alt).median());
+            }
+        }
+        return perAltRankSumResults;
+    }
+
+    /**
+     * Parses expected raw data strings per-allele, which are in the form of run length encoded histograms, and overlays them by
```

Comment:
Boy, that sounds dumb when you put it like that.  Can you think of any way to streamline it?

Reply:
cleaned up the text